zio-http/src/main/scala/zio/http/api/internal/EndpointClient.scala

```diff
@@ -8,11 +8,17 @@ import zio.schema._
 import zio.schema.codec._
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
 
-private[api] final case class EndpointClient[I, O](apiRoot: URL, api: EndpointSpec[I, O]) {
-  def execute(client: Client, input: I)(implicit trace: Trace): ZIO[Any, Throwable, O] = {
+private[api] final case class EndpointClient[I, E, O](apiRoot: URL, api: EndpointSpec[I, E, O]) {
+  def execute(client: Client, input: I)(implicit trace: Trace): ZIO[Any, E, O] = {
```

Comment:
Idea: this could be `HttpError[E]`, encoding decoding errors on a typed error channel. Also it could include response details like status code next to the successfully decoded `E`.

Reply:
Working on it now.